src/librustc_typeck/check/intrinsic.rs

```diff
@@ -303,7 +302,7 @@ pub fn check_intrinsic_type(tcx: TyCtxt<'_>, it: &hir::ForeignItem<'_>) {
                 )
             }
 
-            "try" => {
+            rustc_span::symbol::kw::Try => {
```

Comment:
Good catch. Having `sym::Try` and `kw::Try` mean different things is a bit subtle. I don't see how we could do better though.